Build failure on aarch64 in base/format/common_nodes.h

The flatpak beta aarch64 build of 7841615 failed. The error message was:
```
FAILED: src/base/CMakeFiles/base.dir/format/tree.cc.o 
/usr/bin/c++ -DUSE_XDG -DWL_USE_GLVND -I/run/build/Widelands/src -I/run/build/Widelands/_flatpak_build/src -isystem /usr/include/SDL2 -isystem /app/include/GL -O2 -g -pipe -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -O3 -DNDEBUG  -std=c++11 -O2 -DNDEBUG -DNOPARACHUTE -Wredundant-decls -Wall -Wpedantic -Wdeprecated-declarations -Wextra -Wformat -Wformat-nonliteral -Wformat-security -Wformat-y2k -Winit-self -Winvalid-pch -Wlogical-op -Wmissing-include-dirs -Wno-undef -Wold-style-cast -Woverlength-strings -Wpacked -Wpointer-arith -Wsign-promo -Wsync-nand -Wtrampolines -Wundef -Wunused -Wunused-macros -fno-elide-constructors -Werror -MD -MT src/base/CMakeFiles/base.dir/format/tree.cc.o -MF src/base/CMakeFiles/base.dir/format/tree.cc.o.d -o src/base/CMakeFiles/base.dir/format/tree.cc.o -c /run/build/Widelands/src/base/format/tree.cc
In file included from /run/build/Widelands/src/base/format/wildcard_node.h:24,
                 from /run/build/Widelands/src/base/format/tree.cc:24:
/run/build/Widelands/src/base/format/common_nodes.h: In member function ‘virtual char* format_impl::CharNode::append(char*, format_impl::AbstractNode::ArgType, format_impl::AbstractNode::Argument, bool) const’:
/run/build/Widelands/src/base/format/common_nodes.h:67:34: error: comparison is always true due to limited range of data type [-Werror=type-limits]
   67 |                 if (arg.char_val >= 0 && arg.char_val < 0x20) {
      |                     ~~~~~~~~~~~~~^~~~
```

Build log is temporarily available at https://flathub.org/builds/#/builders/20/builds/5528 (error message is only visible in full log at line 3223 of 3299)

Flatpak manifest at: https://github.com/tothxa/org.widelands.Widelands/commit/c321301a1087a391bc5d58d2a2d54a59ce713b32

--- a/src/base/format/abstract_node.h
+++ b/src/base/format/abstract_node.h
@@ -46,7 +46,7 @@ struct AbstractNode {
 	}
 
 	union Argument {
-		char char_val;
+		signed char char_val;
 		const char* string_val;
 		double float_val;
 		bool boolean_val;

